src/webgpu/api/operation/buffers/map_oom.spec.ts

```diff
@@ -45,15 +46,27 @@ g.test('mapAsync')
         }),
       oom
     );
-    const promise = t.expectGPUError(
-      'validation', // Should be a validation error since the buffer is invalid.
-      () => buffer.mapAsync(write ? GPUMapMode.WRITE : GPUMapMode.READ),
-      oom
-    );
+
+    let promise: Promise<void>;
+    // Should be a validation error since the buffer is invalid.
+    // Unmap abort error shouldn't cause a validation error.
+    t.expectValidationError(() => {
+      promise = buffer.mapAsync(write ? GPUMapMode.WRITE : GPUMapMode.READ);
+    }, oom);
 
     if (oom) {
-      // Should also reject in addition to the validation error.
-      t.shouldReject('OperationError', promise);
+      if (abortError) {
+        // Should reject with abort error because buffer will be unmapped
+        // before validation check finishes.
+        t.shouldReject('AbortError', promise!);
+      } else {
+        // Should also reject in addition to the validation error.
+        t.shouldReject('OperationError', promise!);
+
+        // Wait for validation error before unmap to ensure validation check
+        // ends before unmap.
+        await promise!;
```

Comment:
Since promise rejects, this should throw an exception and fail the test...
and it does, if I change it to use cases instead of subcases (remove `.beginSubcases()`).
Turns out there is a harness error. I put up a fix here, PTAL:
https://github.com/gpuweb/cts/pull/2102